pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -256,3 +259,44 @@ func (r *Reconciler) ValidateHardware(ctx context.Context, log logr.Logger, clus
 
 	return controller.Result{}, nil
 }
+
+// ValidateRufioMachines checks to ensure all the Rufio machines condition contactable is True.
+func (r *Reconciler) ValidateRufioMachines(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "validateRufioMachines")
+
+	kubeReader := hardware.NewKubeReader(r.client)
+	if err := kubeReader.LoadRufioMachines(ctx); err != nil {
+		log.Error(err, "rufio machine check failure")
```

Comment:
I believe `LoadRufioMachines` loads the existing rufio machines to the kubeReader object. Can we change the error log here, something like?
```suggestion
		log.Error(err, "loading existing rufio machines from the cluster")
```